### PR TITLE
Add lesson panel to TutorAid

### PR DESCRIPTION
**Pull request by wpinrui**

### CODE REVIEW

File: src/main/java/tutoraid/ui/LessonCard.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+public class LessonCard extends Card<Lesson> {
+
+    private static final String FXML = "LessonListCard.fxml";
+
+    private static final String LESSON_NAME_LABEL = "Lesson";
+    private static final String TIMING_LABEL = "Timing";
+    private static final String PRICE_LABEL = "Price";
+    private static final String CAPACITY_LABEL = "Capacity";
+    private static final String STUDENTS_LABEL = "Students";
```
Comment (JunWei3112) [inline src/main/java/tutoraid/ui/LessonCard.java:19]: Should these variable names be prefixed with LABEL instead, since associated constants should have a common prefix? e.g. `LABEL_LESSON_NAME`

File: src/main/java/tutoraid/ui/MainWindow.java [hunk 2]
```diff
@@ -117,18 +126,28 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
         studentListPanelPlaceholder.getChildren().clear();
         studentListPanelPlaceholder.getChildren().add(studentListPanel.getRoot());
     }
 
+    /**
+     * Fills up all the lesson's particulars of this window.
+     */
+    void fillLessonCard(boolean viewAll) {
+        lessonListPanel = viewAll
+                ? fullLessonPanel : minimalLessonPanel;
```
Comment (JunWei3112) [inline src/main/java/tutoraid/ui/MainWindow.java:140]: Should `: minimalLessonPanel` be placed on a new line instead?

File: src/main/resources/view/LessonListCard.fxml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          </minWidth>
+        </Label>
+        <Label fx:id="studentName" text="\$studentName" styleClass="cell_big_label" />
+      </HBox>
+      <FlowPane fx:id="tags" />
+      <Label fx:id="lessonName" styleClass="cell_small_label" text="\$studentPhone" />
+      <Label fx:id="timing" styleClass="cell_small_label" text="\$parentName" />
+      <Label fx:id="price" styleClass="cell_small_label" text="\$parentPhone" />
+      <Label fx:id="capacity" styleClass="cell_small_label" text="\$progress" />
+      <Label fx:id="students" styleClass="cell_small_label" text="\$paymentStatus" />
```
Comment (JunWei3112) [inline src/main/resources/view/LessonListCard.fxml:35]: I'm not sure if this will affect the functionality of a Lesson in full view, but should the values of the text fields match those of the id fields?

File: src/main/java/tutoraid/ui/Card.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+
+/**
+ * A generic class to use as an abstraction for card-like objects.
+ * @param <T> The type of entity to be represented using a card
+ */
+public class Card<T> extends UiPart<Region> {
+    public final T t;
```
Comment (JunWei3112) [inline src/main/java/tutoraid/ui/Card.java:11]: Since the scope of this variable is not that small, maybe it is better to use a more descriptive name for this variable?

File: src/test/java/tutoraid/model/lesson/PriceTest.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static tutoraid.model.lesson.Price.isValidPrice;
+
+import org.junit.jupiter.api.Test;
+
+class PriceTest {
```
Comment (Preshita01) [inline src/test/java/tutoraid/model/lesson/PriceTest.java:11]: I was wondering if we need a case to explicitly tests if `isValidPrice` returns `false` when given a negative number? (e.g. -1.01, -2)
